jinahub/encoder/image/clip.py

```diff
@@ -0,0 +1,75 @@
+from typing import Optional, Iterable, Any, List
+
+import numpy as np
+import torch
+from jina import Executor, DocumentArray, requests
+
+import clip
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:min(i + batch_size, len(data))]
+
+class CLIPImageEncoder(Executor):
+    """
+    Encode image into embeddings.
+
+    :param model_name: use clip.available_models() to see all available models: ['RN50', 'RN101', 'RN50x4', 'ViT-B/32']
+    :param device: device to use for encoding ['cuda', 'cpu]
+    :param default_batch_size: fallback batch size in case there is not batch size sent in the request
+    :param default_traversal_path: fallback traversal path in case there is not traversal path sent in the request
+    :param jit: Whether to load the optimized JIT model (default) or more hackable non-JIT model.
+    :param channel_axis: axis of the color channel
+    """
+
+    def __init__(
+            self,
+            model_name: str = 'ViT-B/32',
+            device: Optional[str] = None,
+            default_batch_size: int = 32,
+            default_traversal_path: str = 'r',
+            jit: bool = True,
+            *args, **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        if not device:
+            device = 'cuda' if not device and torch.cuda.is_available() else 'cpu'
```

Comment:
just wonder `if not device` is a redundant check or not?

Reply:
and does this mean if device is not set, and cuda is available, then we use `cuda`? Would it better if we could provide choices for users to choose gpu or cpu?

Reply:
yes, I removed the redundancy. Good finding